src/blob_file_iterator.cc

```diff
@@ -170,8 +170,10 @@ void BlobFileMergeIterator::Next() {
   assert(current_ != nullptr);
   current_->Next();
   if (current_->status().ok() && current_->Valid()) min_heap_.push(current_);
-  current_ = min_heap_.top();
-  min_heap_.pop();
+  if (!min_heap_.empty()) {
```

Comment:
Please add a unit test for this

Reply:
Ok, it's done.

Reply:
oh wow. how this didn't cause data corruption? looks scary. @Connor1996

Reply:
@yiwu-arbug the behavior of `top()` when queue is empty is undefined. So in our environment, the return value may be `nullptr`, so everything works fine. And in macOS, it doesn't so fail.